### PR TITLE
Allow some customizability in the Lark LaTeX parser

### PR DESCRIPTION
**Pull request by wermos**
#### Brief description of what is fixed or changed
As discussed in a meet with @Upabjojr, I added some hooks to the `LarkLaTeXParser` class to allow the user to add some `%extends` and `%override` lines to the grammar to customize its behavior.

#### Other comments
cc @Upabjojr for review.

#### Release Notes

* parsing
  * The Lark-based parser now allows customizability by allowing the user to specify their own LaTeX grammar file.

### CODE REVIEW

File: sympy/parsing/latex/lark/latex_parser.py [hunk 1]
```diff
@@ -9,12 +9,18 @@
...
         grammar_dir_path = os.path.join(os.path.dirname(__file__), "grammar/")
 
         with open(os.path.join(grammar_dir_path, "latex.lark"), encoding="utf-8") as f:
             latex_grammar = f.read()
 
+        if extends is not None:
+            latex_grammar += "\n" + extends
+
+        if overrides is not None:
+            latex_grammar += "\n" + overrides
```
Comment (sylee957) [inline sympy/parsing/latex/lark/latex_parser.py:22]: How about just giving the option for the file path?

Comment (wermos) [reply sympy/parsing/latex/lark/latex_parser.py:22]: We thought of that, and I think the idea was to make it as simple as possible for users, so @Upabjojr suggested doing it like this instead.

**Comment by Upabjojr** [comment]
This PR is missing some test cases.

**Comment by wermos** [comment]
> This PR is missing some test cases.

Can you give me examples of how people might want to use this feature? I mentioned that the nature of the change makes it really hard for end users to use this because they have to understand the internals and instantiate an object of the `LarkLaTeXParser` class with the modficiations they want.

I think @sylee957's idea to just allow the user to use their own grammar file might be better here.

File: sympy/parsing/latex/lark/latex_parser.py (hunk 1 above)
Comment (wermos) [reply sympy/parsing/latex/lark/latex_parser.py]: I changed it to the file path option instead, because it's simpler and gives the user more flexibility.

**Comment by sylee957** [comment]
I think that it is still valid to give some test examples that actually uses `grammar_file`.

**Comment by wermos** [comment]
Where would these test grammar files live? What kind of tests do we want to see? Do you want me to write a grammar file which extends `latex.lark` by adding support for a custom command that might have been defined using `\newcommand`? If so, what kind of command? I don't know about others, but my custom commands often look like `\newcommand{\R}{\mathrm{R}}` which wouldn't be terribly useful to see here. Do you want something else entirely?

**Comment by sylee957** [comment]
We would not need something complicated like newcommand,

However, I would expect simple changes like extending DIV with $\div$ or MUL with $\cdot$ to be possible with using different grammar file.

**Comment by wermos** [comment]
> However, I would expect simple changes like extending `DIV` with $\div$ or `MUL` with $\cdot$ to be possible with using different grammar file.

Both of the things you suggested are already supported in the Lark-based parser. See [here](https://github.com/sympy/sympy/blob/c75049fd4b5659a1c3672f1ce35727fa0c094403/sympy/parsing/latex/lark/grammar/latex.lark#L104) for the $\div$ support. `CMD_DIV` is $\div$. Similarly, see [here](https://github.com/sympy/sympy/blob/c75049fd4b5659a1c3672f1ce35727fa0c094403/sympy/parsing/latex/lark/grammar/latex.lark#L105) for `\cdot`. `CMD_CDOT` is $\cdot$, as one would expect.

**Comment by sylee957** [comment]
Okay, then I’d expect other way of customization, like removing it. 
Or I want to try something like, replacing decimal separator . with comma ,

**Comment by wermos** [comment]
Okay, I think I can make those tests.

Now, where should these new test grammar files live?

**Comment by sylee957** [comment]
It could be inside python test folders

**Comment by sylee957** [comment]
Yes, I don’t think that there is problem of having .lark files in test folder. Pytest does not pick up such files.

**Comment by sylee957** [comment]
Anyway, my suggestion to use the file path is optional.
If there is some technical difficulties about that, we can also just use simple approaches like string concatenation.

(I believe that the transformers also need to be converted with ‘merge_transformers’ such that it can interpret the rules in import)

**Comment by wermos** [comment]
I added tests for the two examples that you suggested. Since both examples were doable with an `%override`, I didn't need to use `merge_transformers`. I used your suggestion of making tempfiles, instead of creating an entirely new files for both cases.

**Pull request merged**